spec/unit/type/rds_instance_spec.rb

```diff
@@ -108,4 +109,41 @@
     end
   end
 
+  it 'backup retention must be an integer' do
+    expect {
+      type_class.new(:name => 'sample', :backup_retention_period => 'Ten')
+    }.to raise_error(Puppet::ResourceError, /backup_retention_period must be an integer/)
+  end
+
+  it 'should be happy with strings for backup_retention_period' do
+    machine = type_class.new(:name => 'sample', :backup_retention_period => 40)
+    expect(machine.property(:backup_retention_period).insync?('40')).to be true
+  end
+
+  it 'should be happy with integers for backup_retention_period' do
+    machine = type_class.new(:name => 'sample', :backup_retention_period => 45)
+    expect(machine.property(:backup_retention_period).insync?(45)).to be true
+  end
```

Comment:
I've fixed up the new tests. I've left the others due to size, although now the test suite as a hole is some-what stable it would be nice to go through a refactor when time permits.